File: java/src/test/java/io/cucumber/gherkin/ParserTest.java
package io.cucumber.gherkin;

import io.cucumber.messages.IdGenerator;
import io.cucumber.messages.types.GherkinDocument;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ParserTest {

    @Test
    void change_default_language() {
        GherkinTokenMatcher matcher = new GherkinTokenMatcher("no");
        IdGenerator idGenerator = new IncrementingIdGenerator();
        Parser<GherkinDocument> parser = new Parser<>(new GherkinDocumentBuilder(idGenerator, "test.feature"));

        GherkinDocument gherkinDocument = parser.parse("Egenskap: i18n support\n", matcher, "test.feature");
        assertEquals("no", gherkinDocument.getFeature().get().getLanguage());
    }
}

File: dart/test/Parser_test.dart
import 'package:cucumber_messages/cucumber_messages.dart' as messages;
import 'package:gherkin/parser.dart';
import 'package:gherkin/language.dart';
import 'package:gherkin/src/ast/messages_gherkin_document_builder.dart';
import 'package:test/test.dart';

void main() {
  test('Change default language', () {
    var languages = loadGherkinLanguagesFromJsonAsset();
    var dialectProvider = GherkinDialectProvider(languages, 'no');
    var matcher = TokenMatcher(dialectProvider);
    var idGenerator = IdGenerator.incrementingGenerator;

    var builder = MessagesGherkinDocumentBuilder(idGenerator);

    var tokenScanner = StringTokenScanner('Egenskap: i18n support\n');

    var parser = Parser<messages.GherkinDocument>(builder);

    var gherkinDocument = parser.parse(tokenScanner, matcher);

    expect('no', gherkinDocument.feature!.language);
  });
}
